### ISSUE TITLE
Gprinstall - problems with missing artifacts

### ISSUE THREAD

**Issue by simonjwright** [AdaCore/gprbuild#38]
With this `Install` package,
```
   package Install is
      for Prefix use "arm-eabi/lib/gnat/stm32f4";
      for Sources_Subdir use "adainclude";
      for Ali_Subdir use "adalib";
      for Lib_Subdir use "adalib";
      for Required_Artifacts (".") use ("runtime.xml");
      for Artifacts ("adalib") use ("adalib/stm32f407-flash.ld--");
      for Install_Project use "false";
   end Install;
```
(**NB** the `adalib/stm32f407-flash.ld--` in the third line from the bottom is deliberately wrong, the actual name is of course `adalib/stm32f407-flash.ld`)
doesn’t result in any warning messages, contrary to the spec in the [GPR Tools User’s Guide section 2.10.10.14, Package Install Attributes](http://docs.adacore.com/live/wave/gprbuild/html/gprbuild_ug/gprbuild_ug/gnat_project_manager.html#package-install-attributes).

If I then make the `Artifacts` into `Required_Artifacts`, I get
```
$ gprinstall -p -P build_runtime.gpr
Install project Build_Runtime

Execution of gprinstall terminated by unhandled exception
raised ADA.IO_EXCEPTIONS.NAME_ERROR : file "/opt/gcc-8.0.1/arm-eabi/lib/gnat/stm32f4/share/gpr/manifests/../../../adainclude/port.c" does not exist
```
basically at `gprinstall-install.adb:3045`, in `Rollback_Manifest` at the call to `Ada.Directories.Delete_File`.

I think the reason for this is that the re-install without deleting the original installation added a second set of entries to the manifest, so we’re trying to delete them twice (`port.c` is the first entry).

You may think that this deserves an issue of its own.

Anyway, deleting the whole installed tree, I see
```
$ gprinstall -p -P build_runtime.gpr
Install project Build_Runtime
gprinstall: error: path does not exist '/users/simon/cortex-gnat-rts/stm32f4/adalib/'
```
which is highly misleading - that path does exist, the problem is that _one or more of the required files on that_ path does not exist.

**Comment by TurboGit** [comment]
Thanks for the detailed analysis.

**Comment by TurboGit** [comment]
This is now fixed. A proper error message is issued. Thanks.

**Comment by simonjwright** [comment]
All good now. Thanks.
